src/KafkaNET.Library/Messages/Message.cs

```diff
@@ -307,26 +410,30 @@ internal static Message ParseFrom(KafkaBinaryReader reader, long offset, int siz
         /// <summary>
         /// Clean up attributes for message, otherwise there is double decompress at kafka broker side.
         /// </summary>
-        internal void CleanMagicAndAttributesBeforeCompress()
+        internal void CleanAttributesBeforeCompress()
         {
             this.Attributes = 0;
```

Comment:
I wasn't completely sure about this change. It looks like based on the comments that someone was experiencing a magic value of 2, and hence was zeroing things out here, but magic value of 2 is an invalid value.